### PR TITLE
Fix importing NeoForge 1.20.1 modpacks from CurseForge

### LINKED ISSUE #2109: Fails to install NeoForge modpack correctly (doesn't correctly install loader in instance)
### Operating System

Windows 11

### Version of Prism Launcher

Prism Launcher 8.0-develop at commit 5d223f6f9359c7b9017583c13c5f538032380b73

### Version of Qt

Qt 6.6.1

### Description of bug

Installing a neoforge modpack will result in a working instance with the mods correctly installed, the loader correctly set, and it will be able to run.

### Steps to reproduce

1. New Instance
2. In Curseforge search for Prominence II [Classic]
3. Download v2.5.0
4. After installation check version tab in settings of instance
5. See below
image

### Suspected cause

_No response_

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by Trial97**
fixes #2109

### CODE REVIEW

File: launcher/modplatform/flame/FlameInstanceCreationTask.cpp [hunk 1]
```diff
@@ -354,6 +354,8 @@ bool FlameCreationTask::createInstance()
         auto id = loader.id;
         if (id.startsWith("neoforge-")) {
             id.remove("neoforge-");
+            if (id.startsWith("1.20.1-"))
+                id = id.mid(7);  // this is a mess for curseforge
```
Comment (TheKodeToad) [inline launcher/modplatform/flame/FlameInstanceCreationTask.cpp]: Can you not use remove here for consistency?

Comment (Trial97) [reply launcher/modplatform/flame/FlameInstanceCreationTask.cpp]: Like this?
```suggestion
                id.remove("1.20.1-");  // this is a mess for curseforge
```

Comment (TheKodeToad) [reply launcher/modplatform/flame/FlameInstanceCreationTask.cpp]: yeah?

**Review by TheKodeToad — APPROVED** [review]

**Comment by TheKodeToad** [comment]
What about export though?

**Comment by Trial97** [comment]
Oh shoot need to update that one too, give me 5min

**Review by DioEgizio — APPROVED** [review]

**Pull request merged**